src/main/java/io/gravitee/integration/api/plugin/IntegrationProviderFactory.java

```diff
@@ -32,5 +33,24 @@ public interface IntegrationProviderFactory<I extends IntegrationProvider> {
      * @param prefix The prefix to use to load the configuration.
      * @return The created integration provider.
      */
-    I createIntegrationProvider(String id, ExchangeConnectorManager exchangeConnectorManager, Configuration environment, String prefix);
+    @Deprecated(forRemoval = true)
+    default I createIntegrationProvider(
+        String id,
+        ExchangeConnectorManager exchangeConnectorManager,
+        Configuration environment,
+        String prefix
+    ) {
+        return createIntegrationProvider(id, environment, prefix);
+    }
+
+    /**
+     * Create an integration provider using Spring Environment and a Prefix to load the configuration
+     * @param id The id of the integration provider.
+     * @param environment The Environment where to load the configuration.
+     * @param prefix The prefix to use to load the configuration.
+     * @return The created integration provider.
+     */
+    default I createIntegrationProvider(String id, Configuration environment, String prefix) {
+        return null;
```

Comment:
I don't get it. To me, an implementation is mandatory for this method as the previous one is deprecated. Why not simply mark this a a Brealing Change?